src/hpoTerm.js

```diff
@@ -69,7 +71,9 @@ HPOTerm.sanitizeID = function(id) {
   temp = temp.replace(/[\)\]]/g, '_J_');
   temp = temp.replace(/[:]/g, '_C_');
   temp = temp.replace(/\,/g, '__COMMA__');
-  return temp.replace(/[^a-zA-Z0-9,;_\-*]/g, '__');
+  // XRegExp handles unicode characters for us, so languages like chinese don't
+  // get filtered out here.
+  return temp.replace(XRegExp('[^,;_\\p{Dash_Punctuation}*\\pL]', 'g'),  '__');
```

Comment:
I'd rather see the \\pL (biggest, most important set), then \\p{Dash_Punctuation} (next grouping), then all of the punctuation chars grouped next to each other. At the moment, it's a bit disorganised. Functionally looks fine though, so am approving.

Reply:
Fair point, I've re-organised things.